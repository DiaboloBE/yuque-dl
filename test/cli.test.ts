import path from 'node:path'
import fs from 'node:fs'
import { fileURLToPath } from 'node:url'
import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import { TestTools } from './helpers/TestTools'

const __dirname = path.dirname(fileURLToPath(import.meta.url))
const cliPath = path.join(__dirname, '../bin/index.js')

let testTools: TestTools

const mdImgReg = /!\[.*?\]\(https*.*?\)/g

describe('yuque-dl CLI', () => {

  beforeEach(() => {
    testTools = new TestTools()
  })

  afterEach(() => {
    testTools.cleanup()
  })

  it('should work', async () => {
    const { stdout, exitCode, stderr } = await testTools.fork(cliPath, [
      'https://www.yuque.com/yuque/eaghk3',
      '-d', '.'
    ])
    expect(exitCode).toBe(0)
    expect(stdout).toContain('√ 已完成')
    const imgDir = path.join(testTools.cwd, '如何从其他工具迁入语雀/img')
    const indexMdPath = path.join(testTools.cwd, '如何从其他工具迁入语雀/index.md')
    expect(fs.existsSync(imgDir)).toBeTruthy()
    const data = fs.readFileSync(indexMdPath).toString()
    expect(data.match(mdImgReg)).toBeFalsy()
    expect(stderr).toBeFalsy()
  })

  it('ignore img should work ', async () => {
    const { stdout, exitCode } = await testTools.fork(cliPath, [
      'https://www.yuque.com/yuque/eaghk3',
      '-d', '.',
      '-i'
    ])
    expect(exitCode).toBe(0)
    expect(stdout).toContain('√ 已完成')
    const imgDir = path.join(testTools.cwd, '如何从其他工具迁入语雀/img')
    expect(fs.existsSync(imgDir)).toBeFalsy()
    const indexMdPath = path.join(testTools.cwd, '如何从其他工具迁入语雀/导入导出功能.md')
    const data = fs.readFileSync(indexMdPath).toString()
    expect(data).toMatchSnapshot()
  })

  it('generate toc list should work ', async () => {
    const { stdout, exitCode } = await testTools.fork(cliPath, [
      'https://www.yuque.com/yuque/eaghk3',
      '-d', '.',
      '--toc',
      '-i'
    ])
    expect(exitCode).toBe(0)
    expect(stdout).toContain('√ 已完成')
    const imgDir = path.join(testTools.cwd, '如何从其他工具迁入语雀/img')
    expect(fs.existsSync(imgDir)).toBeFalsy()
    const indexMdPath = path.join(testTools.cwd, '如何从其他工具迁入语雀/导入导出功能.md')
    const data = fs.readFileSync(indexMdPath).toString()
    expect(data).toMatchSnapshot()
  })
})
