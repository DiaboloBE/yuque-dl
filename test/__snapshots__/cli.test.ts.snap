// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`yuque-dl CLI > generate toc list should work  1`] = `
"# 导入导出功能

- [导入](#%E5%AF%BC%E5%85%A5)
  * [支持的格式](#%E6%94%AF%E6%8C%81%E7%9A%84%E6%A0%BC%E5%BC%8F)
- [导出](#%E5%AF%BC%E5%87%BA)
  * [导出整个知识库](#%E5%AF%BC%E5%87%BA%E6%95%B4%E4%B8%AA%E7%9F%A5%E8%AF%86%E5%BA%93)
  * [导出单篇内容](#%E5%AF%BC%E5%87%BA%E5%8D%95%E7%AF%87%E5%86%85%E5%AE%B9)
    + [语雀文档导出](#%E8%AF%AD%E9%9B%80%E6%96%87%E6%A1%A3%E5%AF%BC%E5%87%BA)
    + [语雀画板导出](#%E8%AF%AD%E9%9B%80%E7%94%BB%E6%9D%BF%E5%AF%BC%E5%87%BA)
    + [语雀数据表导出](#%E8%AF%AD%E9%9B%80%E6%95%B0%E6%8D%AE%E8%A1%A8%E5%AF%BC%E5%87%BA)
    + [语雀表格导出](#%E8%AF%AD%E9%9B%80%E8%A1%A8%E6%A0%BC%E5%AF%BC%E5%87%BA)
- [其他](#%E5%85%B6%E4%BB%96)

---

# 导入


在知识库的新建入口，会有导入功能，点击以后会弹出导入选择框，选择对应的文件按照步骤进行导入。

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695203484501-ef060cfb-9437-4288-9845-69f0ccf9c283.png)





## 支持的格式


语雀的导入功能支持的格式如下：



+ 语雀自有格式
    - 语雀知识库 .lakebook
    - 语雀文档 .lake
    - 语雀画板 .lakeboard .lakemind
    - 语雀数据表 .laketable
    - 语雀表格 .lakesheet



+ 三方格式
    - Word (.docx .doc)
    - Excel（.xlsx .xls .csv）
    - Markdown (.md  .markdown .mark .txt)
    - Evernote (.html + zip)

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695868753252-07cdd364-427c-4caf-8482-6fb2686bd742.png)



# 导出
## 导出整个知识库
如想导出整个知识库，可以在「知识库设置」-「更多设置」中操作：

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695203537507-10a3c0fc-0095-41d0-807f-b52973af3f0d.png)

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695203513367-b52435c3-4c57-4e5d-9ba0-6f33476126a0.png)



支持导出格式：

+ Lakebook（语雀自有格式）
+ PDF





## 导出单篇内容
如想导出单篇内容，可在语雀内容的右上角选择导出：

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1686194300718-10de57e6-3cbf-4e25-8a7c-c27673cf26a4.png)



不同的语雀内容，支持导出的格式不同。具体如下：



### 语雀文档导出
支持导出以下格式：

+ word
+ Markdown
+ PDF
+ JPG

![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786776453-0ac9bec6-ebdd-4e70-aaa1-3a70741f84f3.png)



### 语雀画板导出


单独导出图片步骤：

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1686730073489-0ff4646a-08dc-40c2-b58d-c653a974d06f.png)



支持导出以下格式：

+ JPG
+ PNG



![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786818586-6fbbd02f-64eb-4389-a11a-94df830e9145.png)



### 语雀数据表导出
支持导出以下格式：

+ Excel

![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786831753-bc5d8a32-0669-4ecd-b631-82fac962cd77.png)



### 语雀表格导出
支持导出以下格式：

+ Excel

![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786852594-d6506efe-21ef-4386-9f9d-26b371f7fa97.png)





# 其他


**导入过程会有进度显示，存在有几个状态**：

+ 等待中 - 并发上传数为 2，其他文件将会是等待中状态
+ 上传中 - 正在上传文件，同时在百分比的进度
+ 处理中 - 上传到服务器的文件会到处理中的状态
+ 失败 - 处理失败或者上传失败的状态
+ 成功 - 最终处理完成的状态



如果对语雀有任何的建议或意见，可以提交反馈：[语雀反馈通道](https://www.yuque.com/feedbacks/new)<font style="color:rgb(38, 38, 38);"> ，感谢你的支持！</font><font style="color:rgb(38, 38, 38);">  
</font>

![](https://cdn.nlark.com/yuque/0/2023/png/104907/1679035455225-ceab2fce-01bf-4632-9608-b60a2884a8e7.png?x-oss-process=image%2Fresize%2Cw_1200%2Climit_0%2Fresize%2Cw_1200%2Climit_0)



> 更新: 2023-11-06 09:45:18  
> 原文: <https://www.yuque.com/yuque/eaghk3/ze2dz4>"
`;

exports[`yuque-dl CLI > ignore img should work  1`] = `
"# 导入导出功能

# 导入


在知识库的新建入口，会有导入功能，点击以后会弹出导入选择框，选择对应的文件按照步骤进行导入。

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695203484501-ef060cfb-9437-4288-9845-69f0ccf9c283.png)





## 支持的格式


语雀的导入功能支持的格式如下：



+ 语雀自有格式
    - 语雀知识库 .lakebook
    - 语雀文档 .lake
    - 语雀画板 .lakeboard .lakemind
    - 语雀数据表 .laketable
    - 语雀表格 .lakesheet



+ 三方格式
    - Word (.docx .doc)
    - Excel（.xlsx .xls .csv）
    - Markdown (.md  .markdown .mark .txt)
    - Evernote (.html + zip)

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695868753252-07cdd364-427c-4caf-8482-6fb2686bd742.png)



# 导出
## 导出整个知识库
如想导出整个知识库，可以在「知识库设置」-「更多设置」中操作：

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695203537507-10a3c0fc-0095-41d0-807f-b52973af3f0d.png)

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1695203513367-b52435c3-4c57-4e5d-9ba0-6f33476126a0.png)



支持导出格式：

+ Lakebook（语雀自有格式）
+ PDF





## 导出单篇内容
如想导出单篇内容，可在语雀内容的右上角选择导出：

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1686194300718-10de57e6-3cbf-4e25-8a7c-c27673cf26a4.png)



不同的语雀内容，支持导出的格式不同。具体如下：



### 语雀文档导出
支持导出以下格式：

+ word
+ Markdown
+ PDF
+ JPG

![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786776453-0ac9bec6-ebdd-4e70-aaa1-3a70741f84f3.png)



### 语雀画板导出


单独导出图片步骤：

![](https://cdn.nlark.com/yuque/0/2023/png/34713865/1686730073489-0ff4646a-08dc-40c2-b58d-c653a974d06f.png)



支持导出以下格式：

+ JPG
+ PNG



![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786818586-6fbbd02f-64eb-4389-a11a-94df830e9145.png)



### 语雀数据表导出
支持导出以下格式：

+ Excel

![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786831753-bc5d8a32-0669-4ecd-b631-82fac962cd77.png)



### 语雀表格导出
支持导出以下格式：

+ Excel

![](https://cdn.nlark.com/yuque/0/2022/png/303152/1667786852594-d6506efe-21ef-4386-9f9d-26b371f7fa97.png)





# 其他


**导入过程会有进度显示，存在有几个状态**：

+ 等待中 - 并发上传数为 2，其他文件将会是等待中状态
+ 上传中 - 正在上传文件，同时在百分比的进度
+ 处理中 - 上传到服务器的文件会到处理中的状态
+ 失败 - 处理失败或者上传失败的状态
+ 成功 - 最终处理完成的状态



如果对语雀有任何的建议或意见，可以提交反馈：[语雀反馈通道](https://www.yuque.com/feedbacks/new)<font style="color:rgb(38, 38, 38);"> ，感谢你的支持！</font><font style="color:rgb(38, 38, 38);">  
</font>

![](https://cdn.nlark.com/yuque/0/2023/png/104907/1679035455225-ceab2fce-01bf-4632-9608-b60a2884a8e7.png?x-oss-process=image%2Fresize%2Cw_1200%2Climit_0%2Fresize%2Cw_1200%2Climit_0)



> 更新: 2023-11-06 09:45:18  
> 原文: <https://www.yuque.com/yuque/eaghk3/ze2dz4>"
`;
